### PR TITLE
New deployment config

### PR DESCRIPTION
**Pull request by istankovic**
This PR
- introduces a helper `config` module under `scripts/deployment` that uses `apischema` to validate the loaded JSON config
- in addition to types and structural checks, validation also performs value checks such as whether an address is checksummed or that an expected keyword is used (e.g. `mintable_token`)
- adds a script for Ethereum E2E test
- enables the Ethereum E2E test on CI

### CODE REVIEW

**Comment by bilbeyt** [comment]
general question: for token address, can we somehow validate if it is lets say checksumaddress with this approach ? or maybe with adding it exclusively ?

**Review by bilbeyt — APPROVED** [review]

**Comment by GabrielBuragev** [comment]
> general question: for token address, can we somehow validate if it is lets say checksumaddress with this approach ? 
I would assume that is doable by using regex. If so, we should do that as i faced many problems with different components of our system crashing on non-checksum addresses.

**Review by fredo — APPROVED** [review]
Looks very nice and clean! very good job! 
I'm super happy with the new apischema usage. 

I'm a bit unhappy with the ${resolver} replacement, but I understand that the alternative would come with the tradeoff of special casing the `EthereumL2Messenger`. Let's not spend too much time on this with the upcoming rework of the deployment infra.

**Comment by istankovic** [comment]
> Looks very nice and clean! very good job! I'm super happy with the new apischema usage.

Haha, thanks, glad you like it. I was super lucky to have run into it.
 
> I'm a bit unhappy with the ${resolver} replacement, but I understand that the alternative would come with the tradeoff of special casing the `EthereumL2Messenger`. Let's not spend too much time on this with the upcoming rework of the deployment infra.

Right, I was not entirely happy with it myself but just decided to put that with the rest of the pile on The Mountain  Of Technical Debt.

**Review by GabrielBuragev — APPROVED** [review]
Much much much more readable! Great upgrade with the types and validation.

I have one suggestion. Lets rename the `ganache-local-ethereum.json` to `ethereum-local.json` so we follow the same pattern as for the other rollups. This way the files are more distinguishable between each other.

For the rest, LGTM.

File: scripts/deployment/config.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    lp_fee_ppm: int = field(metadata=schema(min=0))
+    protocol_fee_ppm: int = field(metadata=schema(min=0))
+
+
+@dataclass
+class Chain:
+    name: str
+    rpc: str
+    chain_id: ChainId = field(metadata=schema(min=1))
+    l1_messenger: str | tuple[str, ...]
```
[the remark this answers is not in the document]
Comment (istankovic) [reply scripts/deployment/config.py:47]: > How does the `tuple[str, ...]` type work here?

The Python type annotation
```python
    l1_messenger: str | tuple[str, ...]
```
produces the following JSON schema:
```json
"l1_messenger": {
  "anyOf": [
    {
      "type": "string"
    },
    {
      "type": "array",
      "items": {
        "type": "string"
      }
    }
  ]
},
```
while the following type annotation:
```python
    l1_messenger: str | tuple[str]
```
produces the following JSON schema:
```json
"l1_messenger": {
  "anyOf": [
    {
      "type": "string"
    },
    {
      "type": "array",
      "prefixItems": [
        {
          "type": "string"
        }
      ],
      "items": false,
      "minItems": 1,
      "maxItems": 1
    }
  ]
},
```

The latter essentially specifies a one-element JSON array, which is not what we want in this case.
Please also see https://docs.python.org/3/library/typing.html#typing.Tuple and in particular:

> To specify a variable-length tuple of homogeneous type, use literal ellipsis, e.g. Tuple[int, ...]. A plain [Tuple](https://docs.python.org/3/library/typing.html#typing.Tuple) is equivalent to Tuple[Any, ...], and in turn to [tuple](https://docs.python.org/3/library/stdtypes.html#tuple).

**Pull request merged**